### PR TITLE
Improve PEP-compliance and testing

### PR DESCRIPTION
**Pull request by eigenein**
* Add `tox.ini`, support `2.6`, `2.7`, `3.4`, `3.5`, `3.6` and `3.7`
* Drop Python `3.2` and `3.3` support
* Improve PEP8-compliance in a few places
* Bump version to `1.0.2` and mark it as stable
* Add classifiers to `setup.py`
* Improve `.gitignore` with standard templates for popular environments
* Remove some dead code in comments

### CODE REVIEW

**Comment by eigenein** [comment]
@podshumok Could you please take a quick look if it's okay for you?

File: smpplib/client.py [hunk 1]
```diff
@@ -65,15 +65,11 @@ class Client(object):
...
-        """Initialize"""
-
         self.host = host
         self.port = int(port)
         self._socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self._socket.settimeout(timeout)
-        if sequence_generator is None:
-            sequence_generator = SimpleSequenceGenerator()
-        self.sequence_generator = sequence_generator
+        self.sequence_generator = sequence_generator or SimpleSequenceGenerator()
```
Comment (podshumok) [inline smpplib/client.py]: I think `is None` version is better here as people may want to override `__bool__` for things like sequence generators

**Comment by podshumok** [comment]
Some strings are too long for my screen (I have vision problems and hence I use PEP8 recommended 79 columns). Apart from that it looks good.

File: smpplib/client.py (hunk 1 above)
Comment (eigenein) [reply smpplib/client.py]: Yeah, makes sense. I'll change it back.

**Comment by eigenein** [comment]
Got it, will reformat those lines.

**Comment by eigenein** [comment]
Oki, addressed the comments.

**Pull request merged**